src/circuit/mod.rs

```diff
@@ -449,6 +450,134 @@ impl<P: Poly> PolyCircuit<P> {
         circuit_id
     }
 
+    /// Extract the constant term (coefficient a_i).
+    /// LT isolate sets all non constant coefficients to zero.
+    ///
+    /// This requires a public lookup table that maps each polynomial to its
+    /// constant term version. The lookup_id parameter should be the ID
+    /// returned by register_lt_isolate_lookup.
+    pub fn apply_lt_isolate(&mut self, input: GateId, lookup_id: usize) -> GateId {
+        self.public_lookup_gate(input, lookup_id)
+    }
+
+    /// Register an LT_isolate lookup table for extracting constant terms.
+    /// The table maps each possible polynomial value to a polynomial with only
+    /// the constant term preserved (all other coefficients set to zero).
+    pub fn register_lt_isolate_lookup(&mut self, params: &P::Params, max_value: usize) -> usize {
+        let mut lt_isolate_table = HashMap::new();
+
+        // Create lookup entries for all possible polynomial values
+        for i in 0..=max_value {
+            let input_poly = P::from_usize_to_constant(params, i);
```

Comment:
I think the isolation lookup tables should be defined under a gadget directory because it is just one application of the lookup table.